lib/companies_web/router.ex

```diff
@@ -15,6 +15,10 @@ defmodule CompaniesWeb.Router do
   end
 
   pipeline :auth do
+    plug CompaniesWeb.Plugs.Authenicate
```

Comment:
Instead of a new plug we could update the previous to support a `maintainer: true` option for admin pipeline.

Reply:
I agree it will be better to update the existing plug like:
```elixir
  def call(%{assigns: %{current_user: %{}}} = conn, []) do
    conn
  end

  def call(%{assigns: %{current_user: %{maintainer: true}}} = conn, maintainer: true) do
    conn
  end
```